components/style/sharing/mod.rs

```diff
@@ -109,6 +218,103 @@ pub struct StyleSharingCandidateCache<E: TElement> {
     cache: LRUCache<StyleSharingCandidate<E>>,
 }
 
+/// A struct we use as a simple cache of stuff we use to test against different
+/// candidates.
+#[derive(Debug)]
+pub struct ElementSharingChecker<E: TElement> {
```

Comment:
I'm not really sold on the need for this class. It really seems to me like it's just a bunch of extra boilerplate and indirection, and that we'd be better off by just passing around the element and cache directly.